vault/core.go

```diff
@@ -356,7 +356,7 @@ type Core struct {
 	// baseLogger is used to avoid ResetNamed as it strips useful prefixes in
 	// e.g. testing
 	baseLogger log.Logger
-	logger     log.Logger
+	logger     log.InterceptLogger
```

Comment:
I'm not certain, but is it necessary to change this everywhere? `log.Logger` is an interface which should still be satisfied by  `InterceptLogger`. Wondering if we could keep this as is and just cast the variable to an intercept logger when necessary (in the Monitor object only?). Doing so would also allow for leaving the NewNullLogger calls in the changed test files.  

Also, it seems that the logic for Named and NamedIntercept functions are exactly the same:

https://github.com/hashicorp/go-hclog/blob/1c711f369b4b8d6c4bc07e7e6697e447e13b79ce/interceptlogger.go#L126-L137

https://github.com/hashicorp/go-hclog/blob/1c711f369b4b8d6c4bc07e7e6697e447e13b79ce/interceptlogger.go#L153-L164

Reply:
Yeah, that's a good point.  This is likely an artifact of my shallow knowledge of Go.  I'll give your suggestion a shot.